sqlglot/dialects/snowflake.py

```diff
@@ -46,7 +48,10 @@ def _builder(args: t.List) -> exp.Func:
                     return exp.UnixToTime(this=value, scale=seq_get(args, 1))
                 if not is_float(value.this):
                     return build_formatted_time(exp.StrToTime, "snowflake")(args)
-
+            if kind == exp.DataType.Type.DATE:
+                formatted_exp = build_formatted_time(exp.TsOrDsToDate, "snowflake")(args)
+                formatted_exp.set("safe", safe)
+                return formatted_exp
```

Comment:
We need to be careful here, snowflake has `TO_DATE('<integer>')` - we can't produce a `TsOrDsToDate` for it because it represents a different concept. Check out the logic in the branch right above. I think I'd fall back to producing an anonymous function if we have the integer case for now.

Reply:
Let's add a test for this as well.

Reply:
I will change it so that the `'<integer>'` case falls back to anonymous function. The correct approach would be to map it to `UnixToTime` using the [appropriate scale](https://docs.snowflake.com/en/sql-reference/functions/to_date#usage-notes) but it's not straightforward and the docs mention a deprecation warning.

Reply:
Sounds good